java/jdbcdriver-virtual-threads/src/main/java/com/oracle/dev/jdbc/PipelineVectorDemo.java

```diff
@@ -0,0 +1,789 @@
+            new RuntimeException(
+              "MODEL_OCID must be set as an environment variable"));
+
+      COMPARTMENT_OCID  =
+        Optional.ofNullable(System.getenv("COMPARTMENT_OCID"))
+          .orElseThrow(() ->
+            new RuntimeException(
+              "OCI_COMPARTMENT must be set as an environment variable"));
+    }
+    catch (Exception exception) {
+      exception.printStackTrace();
+      throw new IllegalStateException(exception);
+    }
+  }
+
+  private final static String DB_URL = DatabaseConfig.getDbUrl();
+  private final static String DB_USER = DatabaseConfig.getDbUser();
+  private final static String DB_PASSWORD = DatabaseConfig.getDbPassword();
+
+  public static void main(String[] args) throws Exception {
+    try (
+      Connection connection = DriverManager.getConnection(
+        DB_URL, DB_USER, DB_PASSWORD);
+      Statement statement = connection.createStatement();
+      AutoCloseable dropTable = () -> statement.execute("DROP TABLE example");
+    ) {
+
+      // Create a table with columns to store text and the vector embedding of
+      // the text.
+      statement.execute(
+        """
+          CREATE TABLE example(
+            id NUMBER GENERATED ALWAYS AS IDENTITY,
+            text CLOB NOT NULL,
+            embedding VECTOR(*, FLOAT32),
+            PRIMARY KEY (id))
+          """);
+
+      // Insert text data into the data
+      loadTable(connection);
+
+      // Update the table with embeddings for the text data
+      updateTable(connection);
+
+      // Perform a similarity search for some search terms
+      List<String> searchTerms = List.of(
+        "Predatory behavior of cats",
+        "Location of bears",
+        "Best climate for dogs",
+        "Animals in ancient times",
+        "Beautiful birds",
+        "Deadly fish",
+        "Where the wild things are");
+
+      List<String> searchResults = searchTableText(connection, searchTerms);
+
+      for (int i = 0; i < searchResults.size(); i++) {
+        System.out.println(searchTerms.get(i) + " :\n" + searchResults.get(i));
+        System.out.println();
+      }
+    }
+  }
+
+  /**
+   * Loads the example table with facts about animals. This method uses
+   * the structured concurrency API (preview feature in JDK 22). Calls to
+   * {@link StructuredTaskScope#fork(Callable)} execute concurrent batch INSERTs
+   * into Oracle Database. The batch INSERTs are pipelined: This allows for
+   * concurrent progress of MANY statements from ONE JDBC connection.
+   */
+  static void loadTable(Connection connection)
+    throws IOException, InterruptedException, ExecutionException {
+
+    long start = System.currentTimeMillis();
+
+    // Stream text from a book about animals
+    String bookUrl =
+      "https://www.gutenberg.org/cache/epub/37959/pg37959.txt";
+    try (
+      Stream<String> paragraphStream = streamParagraphs(bookUrl);
+      ShutdownOnFailure taskScope = new ShutdownOnFailure()) {
```

Comment:
To be consistent with the JavaDoc (https://docs.oracle.com/en%2Fjava%2Fjavase%2F21%2Fdocs%2Fapi%2F%2F/java.base/java/util/concurrent/StructuredTaskScope.ShutdownOnFailure.html) can you please declare the scope as a StructuredTaskScope.ShutdownOnFailure?

Reply:
Corrected in latest push. Followed the JavaDoc style of using "var" as well.